flycheck.el

```diff
@@ -6918,8 +6919,40 @@ See URL `http://www.erlang.org/'."
   :error-patterns
   ((warning line-start (file-name) ":" line ": Warning:" (message) line-end)
    (error line-start (file-name) ":" line ": " (message) line-end))
+  :enabled (lambda () (not (flycheck-rebar-config-exists-p)))
   :modes erlang-mode)
 
+
+(flycheck-define-checker erlang-rebar3
+  "An Erlang syntax checker using the rebar3 build tool.
+
+This flychecker compiles the entire rebar3 project, which
+compensates for contextual errors that the regular erlang checker
+wouldn't be aware of. One such example is the erlang -include
+directive. Rebar3 is aware of the context for those includes,
+where the standard erlang checker is not. This checker should be
+used instead of the erlang checker for all rebar3 projects
+
+See URL `https://www.rebar3.org'."
+  :command ("rebar3" "compile")
+  :error-patterns
+  ((warning line-start
+            (file-name) ":" line ": Warning:" (message) line-end)
+   (error line-start
+          (file-name) ":" line ": " (message) line-end))
+  :modes erlang-mode
+  :enabled (lambda () (flycheck-rebar-config-exists-p))
+  :predicate (lambda () (flycheck-buffer-saved-p))
+  :working-directory flycheck-rebar3-project-root)
+
+(defun flycheck-rebar-config-exists-p ()
+  "Whether there is a rebar.config for the current buffer."
+  (locate-dominating-file buffer-file-name "rebar.config"))
```

Comment:
Please remove this extra function, and just use `flycheck-rebar3-project-root` in `:enabled`; you can use `&optional _checker` to be able to invoke the function without specifying a checker.  There's no need to duplicate the same code.

Please also move the function _before_ the definition of the syntax checker.